renderer/src/helpers/hooks/useEditingHistoryManager.ts

```diff
@@ -56,31 +52,32 @@ const undoReducer = <T>(
   }
 };
 
-const useUndoManager = (initialValue: string) => {
-  const [state, dispatch] = useReducer(undoReducer, {
+const useEditingHistoryManager = <T>(initialValue: T) => {
+  const editingHistoryState: EditingHistoryState<T> = {
     undoStack: [initialValue],
     redoStack: [],
-  });
+  };
+  const [state, dispatch] = useReducer(
+    EditingHistoryReducer<T>,
+    editingHistoryState
+  );
 
   const setValue = useCallback(
-    (value) => {
-      dispatch({ type: "SET_VALUE", value });
-    },
+    (value: string) =>
+      dispatch(actions.setValue(value) as EditingHistoryAction<T>),
     [dispatch]
   );
-
   const undo = useCallback(() => {
-    dispatch({ type: "UNDO" });
+    dispatch(actions.undo() as EditingHistoryAction<T>);
 
     if (state.undoStack.length > 2) {
       return state.undoStack[state.undoStack.length - 2];
-    } else {
-      return state.undoStack[0];
     }
-  }, [dispatch, state]);
 
+    return state.undoStack[0];
+  }, [dispatch, state]);
   const redo = useCallback(() => {
-    dispatch({ type: "REDO" });
+    dispatch(actions.redo() as EditingHistoryAction<T>);
```

Comment:
Using `as` is a mistake here. Why do you even cast a `string` to `T`?

Reply:
Resolved